catroid/src/main/java/org/catrobat/catroid/content/bricks/PlaySoundAndWaitBrick.java

```diff
@@ -0,0 +1,376 @@
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
+		final Spinner playSoundAndWaitSpinner = (Spinner) view.findViewById(R.id.playsound_spinner);
+
+		if (!(checkbox.getVisibility() == View.VISIBLE)) {
+			playSoundAndWaitSpinner.setClickable(true);
+			playSoundAndWaitSpinner.setEnabled(true);
+		} else {
+			playSoundAndWaitSpinner.setClickable(false);
+			playSoundAndWaitSpinner.setEnabled(false);
+		}
+
+		if (!(checkbox.getVisibility() == View.VISIBLE)) {
+			playSoundAndWaitSpinner.setOnItemSelectedListener(this);
+		}
+
+		final ArrayAdapter<SoundInfo> spinnerAdapter = createSoundAdapter(context);
+
+		SpinnerAdapterWrapper spinnerAdapterWrapper = new SpinnerAdapterWrapper(context, spinnerAdapter);
+
+		playSoundAndWaitSpinner.setAdapter(spinnerAdapterWrapper);
+
+		setSpinnerSelection(playSoundAndWaitSpinner);
+
+		return view;
+	}
+
+	@Override
+	public View getViewWithAlpha(int alphaValue) {
+
+		if (view != null) {
+
+			View layout = view.findViewById(R.id.brick_play_sound_and_wait_layout);
+			Drawable background = layout.getBackground();
+			background.setAlpha(alphaValue);
+
+			TextView playSoundAndWaitLabel = (TextView) view.findViewById(R.id.brick_play_sound_and_wait_label);
+			playSoundAndWaitLabel.setTextColor(playSoundAndWaitLabel.getTextColors().withAlpha(alphaValue));
+
+			Spinner playSoundAndWaitSpinner = (Spinner) view.findViewById(R.id.playsound_spinner);
+
+			ColorStateList color = playSoundAndWaitLabel.getTextColors().withAlpha(alphaValue);
+			playSoundAndWaitSpinner.getBackground().setAlpha(alphaValue);
+			if (adapterView != null) {
+				((TextView) adapterView.getChildAt(0)).setTextColor(color);
+			}
+
+			this.alphaValue = alphaValue;
+		}
+
+		return view;
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		View prototypeView = View.inflate(context, R.layout.brick_play_sound_and_wait, null);
+
+		Spinner playSoundAndWaitSpinner = (Spinner) prototypeView.findViewById(R.id.playsound_spinner);
+		playSoundAndWaitSpinner.setFocusableInTouchMode(false);
+		playSoundAndWaitSpinner.setFocusable(false);
+		playSoundAndWaitSpinner.setEnabled(false);
+
+		SpinnerAdapter playSoundSpinnerAdapter = createSoundAdapter(context);
+		playSoundAndWaitSpinner.setAdapter(playSoundSpinnerAdapter);
+		setSpinnerSelection(playSoundAndWaitSpinner);
+
+		return prototypeView;
+	}
+
+	private void setSpinnerSelection(Spinner spinner) {
+		if (ProjectManager.getInstance().getCurrentSprite().getSoundList().contains(sound)) {
+			oldSelectedSound = sound;
+			spinner.setSelection(ProjectManager.getInstance().getCurrentSprite().getSoundList().indexOf(sound) + 1, true);
+		} else {
+			if (spinner.getAdapter() != null && spinner.getAdapter().getCount() > 1) {
+				if (ProjectManager.getInstance().getCurrentSprite().getSoundList().indexOf(oldSelectedSound) >= 0) {
```

Comment:
Please add a List<SoundInfo>-variable to assign ProjectManager.getInstance().getCurrentSprite().getSoundList() and use that variable - 
So you don't need to call the entire function all the time.